### PR TITLE
fix: tsconfig includes being a level higher than they should be

### LINKED ISSUE #8872: tsconfig contains duplicate include entries
### Describe the bug

.svelte-kit/tsconfig.json "include" entry looks like this:

```jsonc
{
  "include": [
		"ambient.d.ts",
		"./types/**/$types.d.ts",
		"../vite.config.ts",
		"../src/**/*.js",
		"../src/**/*.ts",
		"../src/**/*.svelte",
		"../src/**/*.js", // duplicate
		"../src/**/*.ts", // duplicate
		"../src/**/*.svelte", // duplicate
		"../tests/**/*.js",
		"../tests/**/*.ts",
		"../tests/**/*.svelte"
	]
}
```

### Reproduction

Go to https://node.new/sveltekit, open .svelte-kit/tsconfig.json

### Logs

_No response_

### System Info

```Shell
-
```

### Severity

annoyance

### Additional Information

_No response_

### LINKED ISSUE #8898: tsconfig.json generated by SvelteKit dev server incorrectly includes parents directory
### Describe the bug

When a custom files.routes option is specified in the svelte.config.js file, the "include" section of the generated tsconfig file in the .svelte-kit folder is instantiated with globs referencing parent directories.

It get entries like this:
```
		"../../../../../**/*.js",
		"../../../../../**/*.ts",
		"../../../../../**/*.svelte",
```
Instead of the expected "../src/**" or "../routes/**". I think this is slowing down my editor and I don't know how to fix this, because the file is autogenerated by default everytime I run the dev server. 

### Reproduction

https://github.com/wvhulle/sveltekit-parent-directory-bug

### Logs

_No response_

### System Info

```Shell
System:
    OS: Linux 5.15 Linux Mint 21.1 (Vera)
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i5-1145G7 @ 2.60GHz
    Memory: 3.22 GB / 15.36 GB
    Container: Yes
    Shell: 3.5.1 - /usr/bin/fish
  Binaries:
    Node: 19.4.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 9.2.0 - /usr/local/bin/npm
  Browsers:
    Chromium: 109.0.5414.74
    Firefox: 109.0
  npmPackages:
    @sveltejs/adapter-auto: ^1.0.0 => 1.0.2 
    @sveltejs/kit: ^1.0.0 => 1.3.10 
    svelte: ^3.54.0 => 3.55.1 
    vite: ^4.0.0 => 4.1.1 
```
```


### Severity

blocking all usage of SvelteKit

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by gtm-nayan**
closes #8872 
closes #8898 

- removed project_relative cause the paths are already resolved to be absolute when the config is processed 
- also fixes the other problem mentioned in https://github.com/sveltejs/kit/issues/8872#issuecomment-1415453987, which was caused by project_relative returning an empty string for any folder directly inside the project directory, so config_relative was pointing to `/**/*.js`
- had to add a special case for src because otherwise it'd be a breaking change

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [ ] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### CODE REVIEW

File: .changeset/happy-boats-shave.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@sveltejs/kit': patch
+---
+
+fix: duplicated paths in tsconfig
```
Comment (dummdidumm) [inline .changeset/happy-boats-shave.md:5]:
```suggestion
fix: deduplicate paths in tsconfig
```

File: packages/kit/src/core/sync/write_tsconfig.js [hunk 2]
```diff
@@ -85,19 +85,28 @@ export function get_tsconfig(kit, include_base_url) {
...
+		'ambient.d.ts',
+		'./types/**/$types.d.ts',
+		config_relative('vite.config.ts')
+	]);
+
+	for (const dir of [
+		kit.files.routes,
+		kit.files.lib,
+		path.resolve('src') // TODO(v2): remove src and only scope it to what's necessary
+	]) {
```
Comment (dummdidumm) [inline packages/kit/src/core/sync/write_tsconfig.js:98]:
```suggestion
	const src_includes = [
		kit.files.routes,
		kit.files.lib,
		path.resolve('src') // TODO(v2): find a better way to include all src files
	].filter((dir) => !path.relative(dir, path.resolve('src')).startsWith('..'));
	for (const dir of src_includes) {
```

File: packages/kit/src/core/sync/write_tsconfig.spec.js [hunk 3]
```diff
@@ -86,4 +86,34 @@ test('Allows generated tsconfig to be replaced', () => {
...
+	const { include } = get_tsconfig(kit, false);
+
+	assert.equal(include, [
+		'ambient.d.ts',
+		'./types/**/$types.d.ts',
+		'../vite.config.ts',
+		'../src/routes/**/*.js',
+		'../src/routes/**/*.ts',
+		'../src/routes/**/*.svelte',
+		'../app/**/*.js',
```
Comment (dummdidumm) [inline packages/kit/src/core/sync/write_tsconfig.spec.js:107]:
```suggestion
		'../app/**/*.js',
```

**Review by dummdidumm — APPROVED** [review]
Makes sense! I adjusted the v2 comment because frankly I don't think we can remove that logic for v2, and deduplicated paths some more

**Comment by wvhulle** [comment]
For me it seems like there are multiple extra levels, not just one. Is this also solved with this pull request?

**Comment by gtm-nayan** [comment]
Yep, see second bullet from the PR description.

File: packages/kit/src/core/sync/write_tsconfig.js (hunk 2 above)
Comment (gtm-nayan) [reply packages/kit/src/core/sync/write_tsconfig.js]: Looks like the check is wrong and the tests are failing. I can never remember the right order of parent/child and whether to use the not for this.

But I think `!path.relative(dir, path.resolve('src')).endsWith('..')` would work,

dir = src/lib/something => "../..",
dir = src/routes => "..",
dir = src => "",
dir = app => "../src"

am I missing any case?

**Comment by wvhulle** [comment]
It is not solved yet on master branch. Do you want to see a reproduction?

**Comment by wvhulle** [comment]
```
[path.resolve('src')].filter((dir) => {
    const relative = path.relative(path.resolve('src'), dir);
    return !relative || relative.startsWith('..');
});
``` 
always returns src even though it doesn't exist. Is this intended?

**Comment by wvhulle** [comment]
```
const config = {
		preprocess: vitePreprocess(),

		kit: {
			env: { dir: "../" },
			alias: {
				$lib: "../lib"
			},
			files: {
				assets: "../static",
				hooks:  { client: './hooks.client', server: './hooks.server' },
				lib: "../lib",
				params: "../lib/params",
				routes: './routes',
				appTemplate: `../static/app.html`
			},
			outDir: `/home/[omitted]/cache/dev/main/kit`,
			adapter: adapter({ out: "build" })
		}
	};
```
generates
 ```
	"include": [
		"ambient.d.ts",
		"./types/**/$types.d.ts",
		"../../../../main/vite.config.ts",
		"../../../../../../../../../**/*.js",
		"../../../../../../../../../**/*.ts",
		"../../../../../../../../../**/*.svelte",
		"../../../../**/*.js",
		"../../../../**/*.ts",
		"../../../../**/*.svelte",
		"../../../../main/tests/**/*.js",
		"../../../../main/tests/**/*.ts",
		"../../../../main/tests/**/*.svelte"
	],
```

**Comment by wvhulle** [comment]
After going through your code it seems like the problem is not in the code adding the includes but in the parsing of the config.

A reproduction is in https://github.com/wvhulle/sveltekit-bug-write-tsconfig.json

**Comment by wvhulle** [comment]
Sorry, I didn't properly update to the latest commit. I now see it works. Reverting to version 1.5.0 brings back the problem. My only remark is that the src is included even though. How do I make sure the kit version i reference is the one from github? I used `pnpm add -D github:sveltejs/kit` but then I had to manually find the path of the function sveltekit in node_modules.

**Pull request merged**